plugins/chromatic_correction.py

```diff
@@ -0,0 +1,222 @@
+            box.ShowModal()
+            box.Destroy()
+            return
+
+        # Calculate Maximum Intensity projection for all the present channels in the stream panel
+        for i, da in enumerate(das):
+            mip_image = model.DataArray(numpy.amax(da, axis=0), copy.copy(da.metadata))
+            dataio.tiff.export(f"{da.metadata[model.MD_DESCRIPTION]}_{i}", mip_image)
+            # All the channels will be corrected according to a reference colored channel
+            # green channel is selected as reference channel, Why? -> Ask Deniz
+            if da.metadata[model.MD_USER_TINT] == (0, 255, 0):
+                self.ref = mip_image
+            else:
+                self.new.append(mip_image)
+
+        # Save and display the reference channel in the analysis tab
+        self.show_static_stream(f"modified_{self.ref.metadata[model.MD_DESCRIPTION]}_ref", self.ref)
+
+        # Compute transformation between reference channel and other channels
+        trans_per_channel = {}  # stores transformation values per channel
+        _, centers_ref = process_image(self.ref)
+        centers1 = numpy.array(centers_ref)
+        for i, im in enumerate(self.new):
+            # Find corresponding points
+            _, centers_new = process_image(im)
+            centers2 = numpy.array(centers_new)
+            max_distance = BEAD_DIAMETER
+            corresponding_pairs = find_corresponding_points(centers1, centers2, max_distance)
+
+            # Extract the matching points
+            points1 = centers1[corresponding_pairs[:, 0]]
+            points2 = centers2[corresponding_pairs[:, 1]]
+
+            if len(corresponding_pairs) > 4:
+                # Estimate the transformation matrix
+                affine2 = AffineTransform.from_pointset(points1, points2)
+                mat = numpy.eye(3)
+                mat[:2, :2] = affine2.matrix
+                mat[:2, 2] = affine2.translation
+                ref_mat = numpy.eye(3)
+                mat = ref_mat @ mat
+
+                # Scale is the norm of the first two columns
+                sx = numpy.linalg.norm(mat[:2, 0])
+                sy = numpy.linalg.norm(mat[:2, 1])
+                scale2 = numpy.array([sx, sy])
+
+                # Compute rotation
+                rotation = numpy.arctan2(mat[1, 0] / sx, mat[0, 0] / sx)
+
+                # compute scale
+                scale = (1 / scale2[0], 1 / scale2[1])
+
+                # compute shear
+                a, b = mat[0, 0], mat[0, 1]
+                c, d = mat[1, 0], mat[1, 1]
+                shear = numpy.arctan2(a * c + b * d, scale[0] * scale[1])
+
+                # compute translation
+                tx = mat[0, 2]
+                ty = mat[1, 2]
+                pixel_size = im.metadata[model.MD_PIXEL_SIZE]
+                translation = (tx * pixel_size[0], -ty * pixel_size[1])
+
+                trans_per_channel[f"channel_{i}"] = {
```

Comment:
I would probably also store some more channel information here, such as the emission/excitation and the reference as well. That way we have all the information to re-create the calibration